tests/openwrt/test_network.py

```diff
@@ -210,6 +211,46 @@ def test_rules_no_src_dest(self):
         )
         self.assertEqual(o.render(), expected)
 
+    def test_render_rule_wrong(self):
+        rule = {
+            "ip_rules": [
+                {
+                    "in": "eth0",
+                    "out": "eth1",
+                    "src": "wrong",
+                    "dest": "wrong1",
+                    "tos": 2,
+                    "action": "blackhole",
+                }
+            ]
+        }
+        o = OpenWrt(rule)
+        with self.assertRaisesRegexp(ValidationError, "'wrong' does not match"):
+            o.validate()
+        rule['ip_rules'][0]['src'] = '192.168.1.1/24'
+        o = OpenWrt(rule)
+        with self.assertRaisesRegexp(ValidationError, "'wrong1' does not match"):
+            o.validate()
+        # fix 'dest' and expect no ValidationError raised
+        rule['ip_rules'][0]['dest'] = '192.168.1.1/24'
+        o = OpenWrt(rule)
+        o.validate()
```

Comment:
please use subtests (or write another test) to ensure the pattern works with several use cases (try at least a few ipv4 and ipv6 addresses)